content/manager/update-deployment.md

```diff
@@ -267,30 +266,227 @@ node_templates:
+```yaml
+# original deployment blueprint
+node_templates:
+    node1:
+        type: Cloudify.nodes.Compute
+    node2:
+        type: my_custom_node_type
+        properties:
+            prop1: value1
+```
 ```yaml
 # deployment update blueprint
 node_templates:
-    node2:  # node1 will be uninstalled. node2 will be installed
+    node1:
+        type: Cloudify.nodes.Compute
+        properties:
+            ip: 192.0.2.1  # modified the property by overriding it default
+    node2:
+        type: my_custom_node_type
+        properties:
+            # removed property prop1
+            prop2: value2  # added property prop2
+```
+#### Outputs
+Outputs can be added, removed or modified.
+```yaml
+# original deployment blueprint
+outputs:
+    output1:
+        value: {get_input: inputA}
+    output2:
+        [...]
+```
+```yaml
+# deployment update blueprint
+outputs:
+    output1:
+        value: {get_input: inputB}  # modified the value of output1
+    # removed output2
+    output3:  # added output3
         [...]
 ```
+#### Workflows
+Workflows can be added, removed or modified.
+```yaml
+# original deployment blueprint
+workflows:
+    workflow1: plugin_name.module_name.task1
+    workflow2:
+        [...]
+```
+```yaml
+# deployment update blueprint
+outputs:
+    workflow1:
+        value: plugin_name.module_name.task2  # modified the value of workflow1
+    # removed workflow2
+    workflow3:  # added workflow3
+        [...]
+```
+
+#### Description:
+The description can be added, removed or modified
+```yaml
+# original deployment blueprint
+# no description field
+```
+```yaml
+# deployment update blueprint
+description: new_description  # added description
+```
+```yaml
+# original deployment blueprint
+description: description_content
+```
+```yaml
+# deployment update blueprint
+# removed the description
+```
+```yaml
+# original deployment blueprint
+description: old_description
+```
+```yaml
+# deployment update blueprint
+description: new_description
+```
 
+### Using a Custom Update Workflow
+If the default `update` workflow doesn't satisfy your every need, Cloudify enables you to create a custom `update` workflow to be used as part of the deployment update process.
+#### Requirements from the custom update workflow
+In order for the custom workflow to be compatible with the deployment update process, it must accept at least the following arguments:
+ * `ctx` - the regular ctx passed to any execution.
+ * `update_id` - the id of the deployment update.
+ * `added_instance_ids` - the list of all the added node instances ids.
+ * `added_target_instances_ids` - the list of all the node instances ids which the added nodes have relationships with.
+ * removed_instance_ids - the list of all the removed node instances id.
+ * remove_target_instance_ids - the list of all the node instances id which the removed nodes had relationships with.
```

Comment:
should this include a target instance which appeared in a removed node but still exists in a non removed node?